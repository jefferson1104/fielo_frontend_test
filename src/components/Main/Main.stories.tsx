import { Story, Meta } from '@storybook/react/types-6-0'
import Main from '.'

export default {
  title: 'Main',
  component: Main
} as Meta

export const Basic: Story = () => (
  <Main
    background={
      'https://images.unsplash.com/32/Mc8kW4x9Q3aRR3RkP5Im_IMG_4417.jpg'
    }
  >
    <h1 style={{ textAlign: 'center', fontSize: '24px', color: '#fff' }}>
      Main Component
    </h1>
  </Main>
)
