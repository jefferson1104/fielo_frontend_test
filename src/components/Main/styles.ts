import styled from 'styled-components'

type WrapperProps = {
  backgroundUrl: string
}

export const Wrapper = styled.main<WrapperProps>`
  height: calc(100vh - 84px);

  background-image: url(${({ backgroundUrl }) => backgroundUrl});
  background-repeat: no-repeat;
  background-size: cover;
`

export const Container = styled.section`
  display: flex;
  align-items: center;
  justify-content: center;

  gap: 30px;
  padding: 100px 0;

  @media only screen and (max-width: 480px) {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    background-color: #222f3e;
  }
`
