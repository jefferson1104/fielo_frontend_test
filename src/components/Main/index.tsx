import { ReactNode } from 'react'
import * as S from './styles'

export type MainProps = {
  background: string
  children: ReactNode
}

const Main = ({ background, children }: MainProps) => {
  return (
    <S.Wrapper backgroundUrl={background}>
      <S.Container>{children}</S.Container>
    </S.Wrapper>
  )
}

export default Main
