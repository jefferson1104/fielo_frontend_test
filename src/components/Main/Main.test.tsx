import { render, screen } from '@testing-library/react'

import Main from '.'

describe('Main component', () => {
  it('should render it correctly', () => {
    const { container } = render(
      <Main background={'http://image-bg.com/example.png'}>
        <h1>Main component</h1>
      </Main>
    )

    expect(
      screen.getByRole('heading', { name: /main component/i })
    ).toBeInTheDocument()

    expect(screen.getByRole('main')).toHaveStyle(
      'background-image: url(http://image-bg.com/example.png)'
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
