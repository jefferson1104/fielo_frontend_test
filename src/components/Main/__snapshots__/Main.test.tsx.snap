// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component should render it correctly 1`] = `
.c0 {
  height: calc(100vh - 84px);
  background-image: url(http://image-bg.com/example.png);
  background-repeat: no-repeat;
  background-size: cover;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 30px;
  padding: 100px 0;
}

@media only screen and (max-width:480px) {
  .c1 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    background-color: #222f3e;
  }
}

<main
  class="c0"
>
  <section
    class="c1"
  >
    <h1>
      Main component
    </h1>
  </section>
</main>
`
